cmd/main.go

```diff
@@ -338,19 +306,27 @@ Full docs can be found at github.com/aws/amazon-` + binName
 		}
 	}
 
-	outputFn := getOutputFn(outputFlag, selector.InstanceTypesOutputFn(resultsOutputFn))
-
-	instanceTypes, itemsTruncated, err := instanceSelector.FilterWithOutput(filters, outputFn)
+	// get filtered instance types
+	instanceTypeDetails, err := instanceSelector.GetFilteredInstanceTypes(filters)
 	if err != nil {
 		fmt.Printf("An error occurred when filtering instance types: %v", err)
 		os.Exit(1)
 	}
-	if len(instanceTypes) == 0 {
+
+	// format instance types as a string
+	maxOutputResults := cli.IntMe(flags[maxResults])
+	instanceTypesString, itemsTruncated, err := instanceSelector.OutputInstanceTypes(instanceTypeDetails, *maxOutputResults, outputFlag)
```

Comment:
Now that we have multiple optional flags influencing output here, what do you think about creating a struct similar to `selector.Filters{}` to encapsulate them? 
While there are only two now, there may be more in the future (i.e. [user specifying which columns to show](https://github.com/aws/amazon-ec2-instance-selector/issues/115#issuecomment-1050146226)), and passing such a struct should mean less changes to the input parameters of functions like `OutputInstanceTypes()` that would care about the new flags. Something like:
```
formatters := OutputFormatters {
    maxOutputResults, 
    outputFlag
}

instanceTypesString, itemsTruncated, err := instanceSelector.OutputInstanceTypes(instanceTypeDetails, formatters)
```

Reply:
I am leaning towards this implementation: https://github.com/aws/amazon-ec2-instance-selector/pull/134#discussion_r909028894

This would mean that `OutputInstanceTypes` would not be a public function anymore. Instead, users would have to call their desired output functions and the truncate function themselves.